autogen/src/dr.rs

```diff
@@ -308,6 +332,221 @@ pub fn gen_dr_operand_kinds(grammar: &[structs::OperandKind]) -> TokenStream {
                 }
             }
         });
+
+        let operand_metadata = kind_to_enum
+            .iter()
+            .filter_map(|(kind, category, enumerators)| {
+                if enumerators.is_empty() {
+                    return None;
+                }
+
+                let mut capability_clauses = BTreeMap::new();
+                let mut extension_clauses = BTreeMap::new();
+                let mut operand_clauses = BTreeMap::new();
+
+                let kind = as_ident(kind);
+                let mut seen_discriminator = BTreeMap::new();
+
+                for e in enumerators {
+                    if seen_discriminator.get(&e.value).is_none() {
```

Comment:
pedantic: you're never reading the name (RHS) of `seen_discriminator`, this could be a `BTreeSet` instead. Then also note that `.insert()` returns `true` if it managed to insert the item, `false` if it already exists. Could be a simple `if seen_descriminator.insert(e.value) { ... }`.